runtime/server/components.go

```diff
@@ -10,14 +10,15 @@ import (
 
 	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
 	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/compilers/rillv1"
 	"github.com/rilldata/rill/runtime/drivers"
 	"github.com/rilldata/rill/runtime/pkg/httputil"
 	"github.com/rilldata/rill/runtime/pkg/observability"
 	"github.com/rilldata/rill/runtime/server/auth"
 	"go.opentelemetry.io/otel/attribute"
 )
 
-func (s *Server) componentDataHandler(w http.ResponseWriter, req *http.Request) error {
+func (s *Server) parseTemplate(w http.ResponseWriter, req *http.Request) error {
```

Comment:
Since the response here is structured and pretty small, it would probably be better to define an RPC with protocol buffers for it